src/dune_engine/fs_cache.ml

```diff
@@ -6,24 +6,31 @@ open Import
 
 type 'a t =
   { name : string (* For debugging *)
-  ; sample : Path.t -> 'a
-  ; cache : 'a Path.Table.t
+  ; sample : Path.Outside_build_dir.t -> 'a
+  ; cache : (Path.Outside_build_dir.t, 'a) Table.t
```

Comment:
Don't we have `Path.Outside_build_dir.Table`? That seems useful.

Reply:
(And then the concern about `Table.t` being slightly slower goes away.)

Reply:
Done.